apps/wallet/src/ui/app/hooks/deepbook/useGetEstimate.ts

```diff
@@ -265,55 +269,82 @@ export function useGetEstimate({
 			baseConversionRate,
 			quoteConversionRate,
 			lotSize,
+			amount,
 		],
 		queryFn: async () => {
-			const [baseCoins, quoteCoins] = await Promise.all([
-				getCoinsByBalance({
+			const sentryTransaction = Sentry.startTransaction({
+				name: 'defi-swap',
+				op: 'get-estimate',
+				data: {
 					coinType,
-					balance: baseBalance,
-					suiClient,
-					address: activeAddress!,
-				}),
-				getCoinsByBalance({
-					coinType,
-					balance: quoteBalance,
-					suiClient,
-					address: activeAddress!,
-				}),
-			]);
-
-			if ((isAsk && !baseCoins.length) || (!isAsk && !quoteCoins.length)) {
-				throw new Error('No coins found in balance');
-			}
-
-			const txn = await getPlaceMarketOrderTxn({
-				deepBookClient,
-				poolId,
-				accountCapId,
-				address: activeAddress!,
-				isAsk,
-				baseCoins,
-				quoteCoins,
-				baseBalance,
-				quoteBalance,
-				walletFeeAddress,
-				totalBaseBalance,
-				totalQuoteBalance,
-				baseConversionRate,
-				quoteConversionRate,
-				lotSize,
+					activeAddress,
+					isAsk,
+					swapAmount: amount,
+					maxAvailableBalanceToSwap: isAsk ? totalBaseBalance : totalQuoteBalance,
+				},
 			});
 
-			if (!accountCapId) {
-				await queryClient.invalidateQueries({ queryKey: ['get-owned-objects'] });
+			try {
+				const [baseCoins, quoteCoins] = await Promise.all([
+					getCoinsByBalance({
+						coinType,
+						balance: baseBalance,
+						suiClient,
+						address: activeAddress!,
+					}),
+					getCoinsByBalance({
+						coinType,
+						balance: quoteBalance,
+						suiClient,
+						address: activeAddress!,
+					}),
+				]);
+
+				if (isAsk ? !baseCoins.length : !quoteCoins.length) {
+					throw new Error('No coins found in balance');
+				}
+
+				const txn = await getPlaceMarketOrderTxn({
+					deepBookClient,
+					poolId,
+					accountCapId,
+					address: activeAddress!,
+					isAsk,
+					baseCoins,
+					quoteCoins,
+					baseBalance,
+					quoteBalance,
+					walletFeeAddress,
+					totalBaseBalance,
+					totalQuoteBalance,
+					baseConversionRate,
+					quoteConversionRate,
+					lotSize,
+				});
+
+				if (!accountCapId) {
+					await queryClient.invalidateQueries({ queryKey: ['get-owned-objects'] });
+				}
+
+				const dryRunResponse = await signer!.dryRunTransactionBlock({ transactionBlock: txn });
+
+				if (dryRunResponse.balanceChanges.length < NUMBER_EXPECTED_BALANCE_CHANGES) {
+					throw new Error('Not enough balance. Please lower swap amount');
```

Comment:
We should probably present a message to the user - can you find out if we can just use a toast here and get confirmation on the wording?